File: java/acronym/src/main/java/Acronym.java
class Acronym {

  String phrase;
  String result = "";

    Acronym(String phrase) {
      if (phrase != null) {
        this.phrase = phrase;
      }
    }

    String get() {
      String word = "";

      while (true) {
        if (phrase.length() == 1) {
          result += word.substring(0,1);
          return result;

        } else if (phrase.substring(0,1).equals(" ")) {
          result += word.toUpperCase().substring(0,1);
          phrase = phrase.substring(1);
          word = "";

        } else {
          word += phrase.substring(0,1);
          phrase = phrase.substring(1);
        }
      }
    }

}

File: csharp/acronym/Acronym.cs
using System;

public static class Acronym
{
    private static string _alphabet = "abcdefghijklmnopqrstuvwxyz";
    private static char[] _seperators = { ' ', '-', ',', '_' };
    public static string Abbreviate(string phrase)
    {
        string[] words = phrase.Split(_seperators, StringSplitOptions.RemoveEmptyEntries);
        string abbreviation = "";

        foreach (var word in words)
        {
            string trimmed = word.Trim(_seperators);
            abbreviation = abbreviation + trimmed[0];
        }

        return abbreviation.ToUpper();
    }
}
